extensions/Worker.Extensions.ServiceBus/src/ServiceBusReceivedMessageConverter.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Azure.Messaging.ServiceBus;
+using Azure.Messaging.ServiceBus.Primitives;
+using Microsoft.Azure.Functions.Worker.Converters;
+using Microsoft.Azure.Functions.Worker.Core;
+
+namespace Microsoft.Azure.Functions.Worker;
+
+internal class ServiceBusReceivedMessageConverter : IInputConverter
+{
+    public ValueTask<ConversionResult> ConvertAsync(ConverterContext context)
+    {
+        if (context is null)
+        {
+            return new ValueTask<ConversionResult>(ConversionResult.Unhandled());
+        }
+
+        return context.Source switch
+        {
+            ModelBindingData binding => ConvertToServiceBusReceivedMessage(context, binding),
+            _ => new ValueTask<ConversionResult>(ConversionResult.Unhandled()),
+        };
+    }
+
+    private ValueTask<ConversionResult> ConvertToServiceBusReceivedMessage(ConverterContext context, ModelBindingData binding)
```

Comment:
Would it be possible to bind to `ServiceBusClient` at some point?

Reply:
Potentially - we will definitely be adding the ability to bind to MessageActions like we do in the in-process model.

Reply:
Not to rush you, I'm really thankful this is finally being worked on, but do you have a rough ETA for this?
Since there's no milestone or anything linked to this PR.